source/DoubleCache.SystemWebCaching/HttpCache.cs

```diff
@@ -35,13 +42,14 @@ public void Add<T>(string key, T item, TimeSpan? timeToLive)
 
         public T Get<T>(string key, Func<T> dataRetriever, TimeSpan? timeToLive) where T : class
         {
-            var item = _cache.Get(key) as T;
-            if (item != null)
-                return item;
-            {
-                item = dataRetriever.Invoke();
-                Add(key, item, timeToLive);
-            }
+            var wrapper = _cache.Get(key) as CacheItemWrapper;
+            if (wrapper != null)
+                return wrapper.Item as T;
```

Comment:
There's an inconsistency introduced here now. Before, and in `MemCache` if you have a key "Users:123" that contains a User object, but you call `Get<Product>("Users:123", () => ...);` then it would see that as a cache miss and call your `dataRetriever` and overwrite the User object with the Product object. Now it will see a cache hit and return null.

The user is doing something incorrect, but do you want this to work the same as before or are you okay with it being different? I suppose in the `RedisCache` implementation it'll throw a `SerializationException`, so it's already not the same everywhere.